### PR TITLE
Move weather data into separate table

### PR DESCRIPTION
**Pull request by acrellin**
This PR moves weather data from the `Telescope` model into a new `Weather` model, with public update access (this addresses the permissions issues we were running into when attempting to update weather info). Handler is updated accordingly, with minor front-end changes.

Closes https://github.com/skyportal/skyportal/issues/1824

weather

### CODE REVIEW

**Comment by acrellin** [comment]
Failure unrelated:
```
FAILED skyportal/tests/api/test_sources.py::test_starlist - Failed: DID NOT R...
```

File: alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
+        'telescopes',
+        sa.Column(
+            'weather_retrieved_at',
+            postgresql.TIMESTAMP(),
```
Comment (kmshin1397) [inline alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py:53]: Should this be of type `sa.DateTime()`?

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Looks good @acrellin, just one comment: when I start from fresh, this is what I see:
image
So I guess the demo data needs to be updated. Otherwise should be good to go.

File: skyportal/models.py [hunk 2]
```diff
@@ -1588,6 +1588,7 @@ class Telescope(Base):
     skycam_link = sa.Column(
         URLType, nullable=True, doc="Link to the telescope's sky camera."
     )
+    weather_link = sa.Column(URLType, doc="Link to the preferred weather site")
```
Comment (profjsb) [inline skyportal/models.py:1591]:
```suggestion
    weather_link = sa.Column(URLType, doc="Link to the preferred weather site.")
```

File: skyportal/models.py [hunk 3]
```diff
@@ -1722,6 +1715,28 @@ def ephemeris(self, time):
         }
 
 
+class Weather(Base):
+    update = public
+
+    weather_info = sa.Column(JSONB, doc="Latest weather information")
```
Comment (profjsb) [inline skyportal/models.py:1721]:
```suggestion
    weather_info = sa.Column(JSONB, doc="Latest weather information.")
```

File: alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        sa.Column('retrieved_at', sa.DateTime(), nullable=True),
+        sa.Column('telescope_id', sa.Integer(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ['telescope_id'], ['telescopes.id'], ondelete='CASCADE'
+        ),
+        sa.PrimaryKeyConstraint('id'),
+    )
+    op.create_index(
+        op.f('ix_weathers_created_at'), 'weathers', ['created_at'], unique=False
+    )
```
Comment (profjsb) [inline alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py:38]: Does this need to be an index?

**Review by profjsb — APPROVED** [review]

File: alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py (hunk 1 above)
Comment (acrellin) [reply alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py:53]: This is what alembic specifies for a `sa.DateTime` column. And the migration checks seem to like this, so seems :+1:

**Comment by acrellin** [comment]
> Looks good @acrellin, just one comment: when I start from fresh, this is what I see:
> image
> So I guess the demo data needs to be updated. Otherwise should be good to go.

@dmitryduev you need a valid weather API key in your local config in order for this to work (look in the fritz-deploy secrets file)

File: alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py (hunk 4 above)
Comment (acrellin) [reply alembic/versions/8f086f8c0c21_move_weather_data_to_separate_table.py:38]: All models that inherit from Base have this indexed column.

**Review by dmitryduev — APPROVED** [review]
Right, thanks :)

**Pull request merged**